Sources/AWSLambdaRuntimeCore/DetachedTasks.swift

```diff
@@ -0,0 +1,169 @@
+        let task = task(context.eventLoop).always { _ in
+            self.storage.remove(key)
+        }
+        self.storage.add(key: key, name: name, task: task)
+        return key
+    }
+    
+    /// Adds a detached async task.
+    ///
+    /// - Parameters:
+    ///   - name: The name of the task.
+    ///   - task: The async task to execute.
+    /// - Returns: A `RegistrationKey` for the registered task.
+    @discardableResult
+    public func detached(name: String, task: @Sendable @escaping () async throws -> Void) -> RegistrationKey {
+        let key = RegistrationKey()
+        let promise = context.eventLoop.makePromise(of: Void.self)
+        promise.completeWithTask(task)
+        let task = promise.futureResult.always { result in
+            switch result {
+            case .success:
+                break
+            case .failure(let failure):
+                self.context.logger.warning(
+                    "Execution of detached task failed with error.",
+                    metadata: [
+                        "taskName": "\(name)",
+                        "error": "\(failure)"
+                    ]
+                )
+            }
+            self.storage.remove(key)
+        }
+        self.storage.add(key: key, name: name, task: task)
+        return key
+    }
+
+    /// Informs the runtime that the specified task should not be awaited anymore.
+    ///
+    /// - Warning: This method does not actually stop the execution of the
+    ///   detached task, it only prevents the runtime from waiting for it before
+    ///   `/next` is invoked.
+    ///
+    /// - Parameter key: The `RegistrationKey` of the task to cancel.
+    public func unsafeCancel(_ key: RegistrationKey) {
+        // To discuss:
+        // Canceling the execution doesn't seem to be an easy
+        // task https://github.com/apple/swift-nio/issues/2087
+        //
+        // While removing the handler will allow the runtime
+        // to invoke `/next` without actually awaiting for the
+        // task to complete, it does not actually cancel
+        // the execution of the dispatched task.
+        // Since this is a bit counter-intuitive, we might not
+        // want this method to exist at all.
+        self.storage.remove(key)
+    }
+
+    /// Awaits all registered tasks to complete.
+    ///
+    /// - Returns: An `EventLoopFuture<Void>` that completes when all tasks have finished.
+    internal func awaitAll() -> EventLoopFuture<Void> {
+        let tasks = storage.tasks
+        if tasks.isEmpty {
+            return context.eventLoop.makeSucceededVoidFuture()
+        } else {
+            return EventLoopFuture.andAllComplete(tasks.map(\.value.task), on: context.eventLoop).flatMap {
+                self.awaitAll()
+            }
+        }
+    }
+}
+
+extension DetachedTasksContainer {
+    /// Lambda detached task registration key.
+    public struct RegistrationKey: Hashable, CustomStringConvertible {
```

Comment:
I think we don't need this. We should not support removing background tasks.

Reply:
This is one of the point I was most worried about, I like the idea of not supporting cancelation at all, it makes things much easier to handle and it is more aligned with how NIO works.